### PR TITLE
Support offset modifier

### PR DESCRIPTION
**Pull request by yeya24**
fixes https://github.com/thanos-community/promql-engine/issues/33

### CODE REVIEW

File: physicalplan/plan.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint time.Time, ma
 		return scan.NewNumberLiteralSelector(model.NewVectorPool(stepsBatch), mint, maxt, step, stepsBatch, e.Val), nil
 
 	case *parser.VectorSelector:
-		filter := scan.NewSeriesFilter(storage, mint, maxt, 0, lookbackDelta, e.LabelMatchers)
+		filter := scan.NewSeriesFilter(storage, mint, maxt, 0, lookbackDelta, e.OriginalOffset, e.LabelMatchers)
```
Comment (fpetkovski) [inline physicalplan/plan.go:45]: Would everything still work if we just adjusted `mint` and `maxt` to avoid propagating this value in different operators?

Comment (yeya24) [reply physicalplan/plan.go:45]: Good point, I think this works.

File: physicalplan/plan.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint time.Time, ma
 			operator := exchange.NewConcurrent(
 				exchange.NewCancellable(
 					scan.NewVectorSelector(
-						model.NewVectorPool(stepsBatch), filter, mint, maxt, step, lookbackDelta, stepsBatch, i, numShards)), 2)
+						model.NewVectorPool(stepsBatch), filter, mint, maxt, step, lookbackDelta, e.OriginalOffset, stepsBatch, i, numShards)), 2)
```
Comment (fpetkovski) [inline physicalplan/plan.go:55]: Can we also do the same mint/maxt adjustment for vector and matrix selectors?

Comment (yeya24) [reply physicalplan/plan.go:55]: No, this offset needs to be propagated anyway since we need to change `o.currentStep` value.

Comment (fpetkovski) [reply physicalplan/plan.go:55]: Got it 👍

**Review by fpetkovski — APPROVED** [review]
Thanks, lgtm

**Pull request merged**